relay-server/src/extractors/signed_json.rs

```diff
@@ -77,7 +78,7 @@ where
     B: axum::body::HttpBody + Send + 'static,
     B::Data: Send,
     B::Error: Into<axum::BoxError>,
-    S: Send + Sync,
+    S: Send + Sync + ServiceRegistry,
```

Comment:
Instead of this trait, you can make this non-generic and require `ServiceState`. This is also what we do in a few other extractors.